arches/app/models/resource.py

```diff
@@ -74,6 +74,18 @@ def __init__(self, *args, **kwargs):
         self.tiles = []
 
     def get_descriptor(self, descriptor, context):
+
+        requested_language = None
+        if context and "language" in context:
+            requested_language = context["language"]
+        language = requested_language or get_language()
+
+        if self.name and self.descriptors:
```

Comment:
I'm not sure if this check is really necessary.  I might just use the try/except block exclusively.  Try to get the value from the model field, if it throws for whatever reason, then recalculate from the function.

Reply:
Ah - good point. Just pushed that change.